### PR TITLE
Add getAllConnectors API to list all registered connectors

### PR DESCRIPTION
**Pull request by tanjialiang**

### CODE REVIEW

File: velox/connectors/Connector.h [hunk 1]
```diff
@@ -397,6 +397,10 @@ bool unregisterConnector(const std::string& connectorId);
 /// Returns a connector with specified ID. Throws if connector doesn't exist.
 std::shared_ptr<Connector> getConnector(const std::string& connectorId);
 
+/// Returns a map of all (connectorId -> connector) pairs currently registered.
```
Comment (xiaoxmeng) [inline velox/connectors/Connector.h:400]: Can you add a unit test for this? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@tanjialiang thanks for adding this.

File: velox/connectors/Connector.h (hunk 1 above)
Comment (mbasmanova) [reply velox/connectors/Connector.h:400]: +1

**Comment by tanjialiang** [comment]
> @tanjialiang Looks good overall, but please add a test and consider renaming the new API to getAllConnectors.
> 
> 
> 
> I'm curious what is the use case which requires such an API.

Thanks Masha, I the use case is more for metrics exporting. We'd like to iterate and dynamically export all registered connectors' metrics. I'll add some tests and update the PR

File: velox/connectors/tests/ConnectorTest.cpp [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  const int32_t numConnectors = 10;
+  for (int32_t i = 0; i < numConnectors; i++) {
+    registerConnector(
+        std::make_shared<TestConnector>(fmt::format("connector-{}", i)));
+  }
+  const auto& connectors = getAllConnectors();
+  EXPECT_EQ(connectors.size(), numConnectors);
+  for (int32_t i = 0; i < numConnectors; i++) {
+    EXPECT_NE(
+        connectors.find(fmt::format("connector-{}", i)), connectors.end());
```
Comment (xiaoxmeng) [inline velox/connectors/tests/ConnectorTest.cpp:62]:
```
nit: std::contains()
```

File: velox/connectors/tests/ConnectorTest.cpp [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  }
+  for (int32_t i = 0; i < numConnectors; i++) {
+    unregisterConnector(fmt::format("connector-{}", i));
+  }
```
Comment (xiaoxmeng) [inline velox/connectors/tests/ConnectorTest.cpp:66]: Can you check getAllConnectors returns empty here? thanks!

**Review by xiaoxmeng — APPROVED** [review]
@tanjialiang  LGTM. Thanks for adding the test!

File: velox/connectors/CMakeLists.txt [hunk 4]
```diff
@@ -11,7 +11,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-add_library(velox_connector Connector.cpp)
+add_library(velox_connector Connector.cpp tests/ConnectorTest.cpp)
```
Comment (kgpai) [inline velox/connectors/CMakeLists.txt:14]: You are adding a test dependency to the connector library ? 
Is there a reason why ?

Comment (tanjialiang) [reply velox/connectors/CMakeLists.txt]: Thanks Krishna I'm removing it

**Pull request closed without merging**